lib/jxl/enc_ac_strategy.cc

```diff
@@ -44,6 +44,13 @@
 // sensitive to some kind of degradation. Unfortunately image quality
 // is still more of an art than science.
 
+// Set JXL_DEBUG_AC_STRATEGY to 1 to enable debugging.
+#ifndef JXL_DEBUG_AC_STRATEGY
+#define JXL_DEBUG_AC_STRATEGY 0
+#endif
+
+#if JXL_DEBUG_AC_STRATEGY
```

Comment:
Let's not use `#if` for this purpose

Reply:
What is the cleaner way to make sure these debug functions (`TypeColor()`, `TypeMask()`, `DumpAcStrategy()`) do not end up getting compiled? Define them anyway, put the call to `DumpAcStrategy()` behind an `if (constexpr) {}` and hope the compiler will remove the dead code?

Reply:
yes, the compiler is usually pretty good at doing that (at least if the functions are in an anonymous namespace), and that ensure that the function will still compile when needed.

Reply:
Makes sense, updated it.
These functions are in `namespace jxl`, should that be updated to an anonymous one?

Reply:
I don't really see why that debug code was in `namespace jxl` in the first place, so just turned it into an anonymous namespace.

Reply:
Oops, the `namespace jxl` was there for a reason after all, made it a nested anonymous namespace.